coreneuron/network/multisend_setup.cpp

```diff
@@ -118,39 +118,37 @@ static int* newintval(int val, int size) {
     return x;
 }
 
-static int* newoffset(int* acnt, int size) {
-    int* aoff = new int[size + 1];
+static std::vector<int> newoffset(const std::vector<int>& acnt) {
+    std::vector<int> aoff(acnt.size() + 1);
     aoff[0] = 0;
-    for (int i = 0; i < size; ++i) {
+    for (int i = 0; i < acnt.size(); ++i) {
         aoff[i + 1] = aoff[i] + acnt[i];
     }
     return aoff;
 }
 
 // input scnt, sdispl ; output, newly allocated rcnt, rdispl
-static void all2allv_helper(int* scnt, int*& rcnt, int*& rdispl) {
+static void all2allv_helper(const std::vector<int>& scnt, std::vector<int>& rcnt, std::vector<int>& rdispl) {
     int np = nrnmpi_numprocs;
-    int* c = newintval(1, np);
-    rdispl = newoffset(c, np);
-    rcnt = newintval(0, np);
-    nrnmpi_int_alltoallv(scnt, c, rdispl, rcnt, c, rdispl);
-    delete[] c;
-    delete[] rdispl;
-    rdispl = newoffset(rcnt, np);
+    std::vector<int> c(np, 1);
+    rdispl = newoffset(c);
+    rcnt = std::vector<int>(np, 0);
+    nrnmpi_int_alltoallv(scnt.data(), c.data(), rdispl.data(), rcnt.data(), c.data(), rdispl.data());
+    rdispl = newoffset(rcnt);
 }
 
 #define all2allv_perf 1
 // input s, scnt, sdispl ; output, newly allocated r, rcnt, rdispl
 static void
-all2allv_int(int* s, int* scnt, int* sdispl, int*& r, int*& rcnt, int*& rdispl, const char* dmes) {
+all2allv_int(const std::vector<int>& s, const std::vector<int>& scnt, const std::vector<int>& sdispl, std::vector<int>& r, std::vector<int>& rcnt, std::vector<int>& rdispl, const char* dmes) {
 #if all2allv_perf
     double tm = nrn_wtime();
 #endif
     int np = nrnmpi_numprocs;
     all2allv_helper(scnt, rcnt, rdispl);
-    r = newintval(0, rdispl[np]);
+    r = std::vector<int>(np, 0);
 
-    nrnmpi_int_alltoallv(s, scnt, sdispl, r, rcnt, rdispl);
+    nrnmpi_int_alltoallv(s.data(), scnt.data(), sdispl.data(), r.data(), rcnt.data(), rdispl.data());
```

Comment:
Not for now, but to put into an issue, the wrappers can be a little more high-level and accept vectors and hide the data-unwrapping `.data()` inside.

Reply:
Not sure... If it's use by someone outside, it has to stay as C API.

Reply:
good point, come to think of it - the modls could use this (and they are not c++ified yet). OK, anyway, for now, no change, but maybe still worth creating an issue.